daemon/cmd/daemon_main.go

```diff
@@ -1855,7 +1856,9 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 			// (re-)allocate the restored identities before we release them.
 			time.Sleep(option.Config.IdentityRestoreGracePeriod)
 			log.Debugf("Releasing reference counts for %d restored CIDR identities", len(d.restoredCIDRs))
-			d.ipcache.ReleaseCIDRIdentitiesByCIDR(d.restoredCIDRs)
+			for cidr := range d.restoredCIDRs {
+				d.ipcache.RemoveMetadata(cidr, restoredCIDRResource, ipcachetypes.RestoredIdentity(0))
+			}
```

Comment:
nit, Ah we probably should figure out better batch APIs for this. Could be hundreds/thousands of lock/edit/unlock and triggering of next label injection. I think the behaviour should be OK in terms of ratelimits and so on but it'd be nice to make all the changes at once.

Reply:
Sure, we should add batched `UpdateMetadataMulti()` and `RemoveMetadataMulti()` methods

Reply:
@joestringer take a look, tell me what you think of the API. It's... fine.